examples/chip-tool/commands/delay/BusyWaitCommand.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "BusyWaitCommand.h"
+
+CHIP_ERROR BusyWaitCommand::RunCommand()
+{
+    auto & clock = chip::System::SystemClock();
+    auto start   = clock.GetMonotonicTimestamp();
+    chip::System::Clock::Milliseconds32 durationInMs(mDurationInMs);
+    while (clock.GetMonotonicTimestamp() - start < durationInMs)
+    {
```

Comment:
how about at least a delayMS of 1 ms here? full 100% CPU busy look sounds really painful.

Also this really needs a comment as to why this is done in such a way or what uses busy wait has in general (so I can review it without looking at the PR history)

Reply:
@andy31415 A `delayMS` would be async and would not block message processing.  The point of this command is as a fault-injection tool: it blocks the event loop for some period of time, causing timeouts on the other side of the communication.

That said, we may be able to just `sleep()` or something instead of busy-waiting.  But I recall something failing when I tried that in the existing places where we have this sort of busy-wait fault-injection....

Reply:
As Boris pointed out, the goal of this command is to block event loop processing, which is why is should not be async.

But this is right that I have not tried sleeping, as I remember it was not working too. Indeed, it does work for my purpose and I don't need to consume the whole CPU.

Reply:
Please add a comment here saying sleep using std is on purpose to block the main loop processing as a fault injection tool.

Reply:
This is all done now.